etc/init.d/zfs-functions.in

```diff
@@ -294,12 +294,17 @@ checksystem()
 	# Just make sure that /dev/zfs is created.
 	udev_trigger
 
-	if ! [ "$(uname -m)" = "x86_64" ]; then
-		echo "Warning: You're not running 64bit. Currently native zfs in";
-		echo "         Linux is only supported and tested on 64bit.";
-		# should we break here? People doing this should know what they
-		# do, thus i'm not breaking here.
-	fi
+	case "$(uname -m)" in
+		aarch64|ppc64|ppc64le|x86_64)
```

Comment:
what about this?

~~~
case "$(uname -m)" in
        *64*)
                ;;
        *)
~~~

Reply:
Actually, these days thing work pretty reasonably even on 32-bit arches (and some more exotic ones like s390x).  I think it would be reasonable to remove this very old warning entirely.